### DISCUSSION TITLE
antd 5 Input 使用 组件token 和 全局 自定义主题有部分样式 异常（token 覆盖不足）。

### DISCUSSION THREAD

**Discussion by x011223** [ant-design/ant-design#46458] [Q&A]
有 allowClear时 背景色异常 和 图标颜色异常，图标颜色需要使用class覆盖样式，但是 hover 颜色又要覆盖，这样的话越来越多的东西通过样式处理，没有找到相关的 token 可以解决。

[demo 在这](https://codesandbox.io/p/sandbox/zhi-shi-tiao-antd-5-12-2-forked-8hlf6x)

![image](https://github.com/ant-design/ant-design/assets/31724313/fd79d8dc-f71f-48c7-b54d-df69ded6c459)

除了样式覆盖有别的方式处理吗？或者token是我没找全？

**Comment by arvinxx** [comment]
这个明显是组件实现问题。 @MadCcc 啥时候修下？上次我就提过了

**Comment by x011223** [comment]
![image](https://github.com/ant-design/ant-design/assets/31724313/5d4444ae-d83d-46eb-8bf0-3d4096ef1ce5)
![image](https://github.com/ant-design/ant-design/assets/31724313/091d0a9e-3990-4ac3-8f53-ba3a42671c60)

DatePicker InputNumber 的输入框 icon 也有问题，应该是同一个问题

**Reply by x011223** [reply]
![image](https://github.com/ant-design/ant-design/assets/31724313/601cda6d-4362-458a-909a-c3d0d2915495)


DatePicker cellHoverWithRangeBg  hover 背景色 （图中 8 号） 应该和 选中色（图中 1 号）一样，文本在背景色中处于居中的位置？？？。

但是这样的话，图中 31 号选中后，看起来又不协调，在我看来只能二选一了。
